modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -4350,6 +4358,51 @@ public void testPartialUpdateObjectEntryWithLocalizedAttachmentObjectField()
 		objectDefinitionLocalService.deleteObjectDefinition(objectDefinition);
 	}
 
+	@Test
+	public void testPartialUpdateObjectEntryWithReadOnlyFields()
+		throws Exception {
+
+		ObjectEntry objectEntry = _defaultObjectEntryManager.addObjectEntry(
+			dtoConverterContext, _objectDefinition2,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						"attachmentObjectFieldName",
+						_getAttachmentObjectFieldValue()
+					).put(
+						"picklistObjectFieldName", _addListTypeEntry()
+					).put(
+						"textObjectFieldName", "Foo"
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		Object attachmentObject = objectEntry.getPropertyValue(
```

Comment:
Can you try to inline this instead of creating two new variables?

Another option would be to keep the created values ​​in variables to use them when you try to update the values, for example:
```
String listTypeEntryKey = _addListTypeEntry();
Long fileEntryId = _getAttachmentObjectFieldValue();
```

Reply:
Unfortunately isn't possible we use these values inline since for use it inside the lambda the value should be "final". And about the second option isn't feasible too since we need to pass the entire object not only the ids to replicate the correct scenario.